src/atoms/SegmentControl/SegmentControlOption.native.js

```diff
@@ -0,0 +1,239 @@
+  const context = useSegmentControlContext();
+  const theme = useContext(ThemeContext);
+  const size = context.size;
+
+  const selected = isSelected({ context, value });
+
+  const onPress = useCallback(() => {
+    if (isDefined(context.onChange)) {
+      context.onChange(value);
+    }
+  }, [context, value]);
+
+  if (size === 'small' && isDefined(subText)) {
+    throw new Error(`SegmentControl\n \`subText\` cannot be used with \`size='small'\``);
+  }
+
+  if (size === 'small' && isDefined(icon)) {
+    throw new Error(`SegmentControl\n \`icon\` cannot be used with \`size='small'\``);
+  }
+
+  return (
+    <Flex flex={1} alignItems="center" justifyContent="center">
+      <Space padding={[1, 1, 1, 1]}>
+        <Size minWidth={makePxValue(6)}>
+          <StyledOption
+            onPress={onPress}
+            underlayColor={styles.option.underlayColor({ theme, selected })}
+            variant={context.variant}
+            selected={selected}
+            disabled={disabled}
+            {...automation(testID)}
+          >
+            <React.Fragment>
+              <Flex flexDirection="column">
+                <View>
+                  <Flex flexDirection="row" alignItems="center">
+                    <View>
+                      {isDefined(icon) && size !== 'small' ? (
+                        <Space margin={[0, 0.5, 0, 0]}>
+                          <View>
+                            <Icon
+                              name={icon}
+                              fill={styles.text.color({ selected, disabled })}
+                              size={styles.icon.size({ size })}
+                            />
+                          </View>
+                        </Space>
+                      ) : null}
+                      <Text
+                        size={styles.text.size({ size })}
+                        color={styles.text.color({ selected, disabled })}
+                        maxLines={1}
+                        _weight={styles.text.weight({ variant: context.variant })}
+                      >
+                        {value}
+                      </Text>
+                    </View>
+                  </Flex>
+                  {isDefined(subText) && size !== 'small' ? (
+                    <View>
+                      <Space padding={isDefined(icon) ? [0, 0, 0, 2.5] : [0, 0, 0, 0]}>
+                        <View>
+                          <Text
+                            size="xsmall"
+                            color={styles.subText.color({ selected, disabled })}
+                            maxLines={1}
+                          >
+                            {subText}
+                          </Text>
+                        </View>
+                      </Space>
+                    </View>
+                  ) : null}
+                </View>
+              </Flex>
+              <Divider hide={hideDivider} />
```

Comment:
Can't render null since it will make keep changing the widths.

Reply:
What? No. How? When you hide it do you mean that you reserve the place? Then it's not proper way of implementing

Reply:
Oh to my surprise you are actually rendering a divider. Why not just have right border?

Reply:
Its not a border. It's a divider that is vertically in the center. There's nothing like borderRightTopMargin :p

Reply:
```!hideDivider ? <Divider/> : null```

I'm doing this. Realized that since my Divider is ```position: absolute``` it'll not take up physical space. and I can hide it based on this.